test/powershell/Language/Parser/RedirectionOperator.Tests.ps1

```diff
@@ -45,42 +45,36 @@ Describe "Redirection operator now supports encoding changes" -Tags "CI" {
         }
     }
 
-    # $availableEncodings = "unknown","string","unicode","bigendianunicode","utf8","utf7", "utf32","ascii","default","oem"
-    $availableEncodings = (Get-Command Out-File).Parameters["Encoding"].Attributes.ValidValues
-
+    $availableEncodings = 
+        @([System.Text.Encoding]::ASCII,
+          [System.Text.Encoding]::BigEndianUnicode,
+          (New-Object System.Text.UTF32Encoding($true,$true)),
+          [System.Text.Encoding]::Unicode,
+          [System.Text.Encoding]::UTF7,
+          [System.Text.Encoding]::UTF8,
+          [System.Text.Encoding]::UTF32);
+              
     foreach($encoding in $availableEncodings) {
-        $skipTest = $false
-        if ($encoding -eq "default") {
-            # [System.Text.Encoding]::Default is exposed by 'System.Private.CoreLib.dll' at
-            # runtime via reflection. However,it isn't exposed in the reference contract of
-            # 'System.Text.Encoding', and therefore we cannot use 'Encoding.Default' in our
-            # code. So we need to skip this encoding in the test.
-            $skipTest = $true
-        }
 
         # some of the encodings accepted by Out-File aren't real,
         # and Out-File has its own translation, so we'll
         # not do that logic here, but simply ignore those encodings
         # as they eventually are translated to "real" encoding
-        $enc = [System.Text.Encoding]::$encoding
-        if ( $enc )
-        {
-            $msg = "Overriding encoding for Out-File is respected for $encoding"
-            $BOM = $enc.GetPreamble()
-            $TXT = $enc.GetBytes($asciiString)
-            $CR  = $enc.GetBytes($asciiCR)
-            $expectedBytes = .{ $BOM; $TXT; $CR }
-            $PSDefaultParameterValues["Out-File:Encoding"] = "$encoding"
-            $asciiString > TESTDRIVE:/file.txt
-            $observedBytes = Get-Content -AsByteStream TESTDRIVE:/file.txt
-            # THE TEST
-            It $msg -Skip:$skipTest {
-                $observedBytes.Count | Should -Be $expectedBytes.Count
-                for($i = 0;$i -lt $observedBytes.Count; $i++) {
-                    $observedBytes[$i] | Should -Be $expectedBytes[$i]
-                }
+        $encodingName = $encoding.EncodingName
+        $msg = "Overriding encoding for Out-File is respected for $encodingName"
+        $BOM = $encoding.GetPreamble()
+        $TXT = $encoding.GetBytes($asciiString)
+        $CR  = $encoding.GetBytes($asciiCR)
+        $expectedBytes = .{ $BOM; $TXT; $CR }
```

Comment:
Not sure why the original test was written this way, this is a bit expensive just to essentially put an array together. Better to use a subexpression here as well:

```suggestion
        $expectedBytes = @( $BOM; $TXT; $CR )
```

You can use line breaks instead of semicolons if you prefer.

Reply:
Made these adjustments in the latest commit.